proposals/LP0004-project-governance.md

```diff
@@ -0,0 +1,393 @@
+
+The proposal process described in [LP0001 LLVM Decision Making](https://github.com/llvm/llvm-www/blob/main/proposals/LP0001-LLVMDecisionMaking.md) should be adjusted in the following ways under this governance proposal.
+
+In step [3](https://github.com/llvm/llvm-www/blob/main/proposals/LP0001-LLVMDecisionMaking.md?plain=1#L60) of the process, in addition to identifying review managers, the proposal will identify the relevant _area team_ to oversee the proposal. If there is no relevant _area team_, or if the decision spans across the domains of multiple _area teams_ the _project council_ will oversee the proposal.
+
+In steps [4](https://github.com/llvm/llvm-www/blob/main/proposals/LP0001-LLVMDecisionMaking.md?plain=1#L61) and [8](https://github.com/llvm/llvm-www/blob/main/proposals/LP0001-LLVMDecisionMaking.md?plain=1#L65), the roles designated to Chris Lattner are replaced by the overseeing _area team_ or the _project council_.
+
+## Voting Process
+
+Public notice of any vote will be published on the LLVM discussion forums under the "Elections" topic at least 14 days before voting opens with a schedule for the election. Each election will be open for 14 days. Reminders will be posted on the "Elections" topic at:
+
+* 7 days before voting opens.
+* The day voting opens.
+* 7 days after voting opens.
+* 48 hours before voting closes.
+* 24 hours before voting closes.
+
+Voter eligibility will be determined 7 days before voting opens by the rules for defining _voting contributors_. Voting will be executed using a private poll hosted by [Concordcet Internet Voting Service]([https://civs1.civs.us/](https://civs1.civs.us/)). Concordcet uses a ranked choice polling process.
+
+All eligible _voting contributors_ will be contacted via email with instructions on how to vote on the day voting opens.
+
+### Elections
+
+Elections for _area teams_ will be conducted in January of each year. A two week long nomination period will begin the second Monday in January. During the nomination period any _voting contributor_ can nominate themselves or another _voting contributor_ to run for any one _area team_. No individual can run for more than one _area team_ in a single election. Nominations will be recorded publicly for community visibility.
+
+Voting begins the fourth Monday in January and continues for 2 weeks. Election results will be announced no later than the second Monday in February. The term of the newly elected _area team_ begins the first Monday in March. Each _area team_ will meet during the first week in March to elect from themselves the team chair to re-constitute the _project council_.
+
+### Mandatory Voting
+
+To encourage participation in elections voting is considered mandatory. Any _eligible voting contributor_ who does not vote in an election will be ineligible to vote in the next election and their registration will be removed if they do not vote in the next election they are eligible to vote in.
```

Comment:
I don't like the idea of mandatory voting. I expect the "eligible voters" base is going to be quite diverse, ranging from people who make two or three commits a year, to people who make more than a thousand. The former kind of contributor may not be interested in voting (e.g. because they don't feel familiar enough with the community to vote for any particular person). And the former kind of contributor may become the latter kind from one year to the next (e.g. going from hobby contribution to an LLVM-centric job) and it seems weird to not allow them to vote just because they did not previously.

Reply:
I'm definitely against mandatory voting. I work on LLVM only during my working hours, and never check in at weekends/during time off etc. 14 days is a small enough window for elections that there will be plenty of people like me who will not be online during the election window because they'll be on vacation or similar. Indeed, I had 6 weeks off straight earlier this year in mid January, so would have missed any such election held this year. Maternity leave in the UK meanwhile is 12 months, and I'm sure there are other circumstances where people would need to have equal or greater periods of time off.

Reply:
+1 to removing mandatory voting from this iteration. The metric for being a member of the community in good standing *has* to be based on contributions. There are plenty of valid reasons that an individual may choose to not vote, including forms of conscientious objector stands, and feelings of psychological safety.

In practice, it seems like a really bad idea to roll out something as sweeping as this policy and have mandatory voting on day one. Increasing voting engagement seems more like a year three problem. Better to use carrots to get something like this off the ground vs sticks. We don't want to find ourselves in a situation a few years from now where the only members eligible to vote are the ones who are naturally biased to engage heavily in processes like this. Let that grow organically and bias towards inclusion.

Reply:
Will remove. This was a late addition to the draft based on earlier feedback. I personally like the idea as a way to encourage contribution, but can understand the concerns.

Reply:
Coming from a country that has mandatory votes, this brings me many bad memories from the dictatorship era. I would not feel comfortable if this remained in the text. Thanks for removing it.